R/truncate_linelist.R

```diff
@@ -59,71 +43,84 @@
 #' linelist <- sim_linelist()
 #' linelist_trunc <- truncate_linelist(linelist)
 #'
-#' # set maximum date to apply truncation to 2023-01-01
-#' linelist_trunc <- truncate_linelist(linelist, max_date = "2023-01-01")
-#'
-#' # apply longer truncation to hospital admission
+#' # set truncation point 3 weeks before the end of outbreak
 #' linelist_trunc <- truncate_linelist(
 #'   linelist,
-#'   delay = function(x) rlnorm(n = x, meanlog = 2, sdlog = 0.5),
-#'   truncation_event = "admission"
+#'   truncation_time = 3,
+#'   unit = "weeks"
 #' )
 #'
-#' # variable right truncation with mean 2 and sd 1 (default behaviour)
+#' # set truncation point to 2 months since the start of outbreak
 #' linelist_trunc <- truncate_linelist(
 #'   linelist,
-#'   delay = function(x) rlnorm(n = x, meanlog = 0.58, sdlog = 0.47)
+#'   truncation_time = 2,
+#'   unit = "months",
+#'   direction = "forwards"
 #' )
 #'
-#' # fixed right truncation of 5 days
+#' # set truncation point to 2023-03-01
 #' linelist_trunc <- truncate_linelist(
 #'   linelist,
-#'   delay = function(x) rep(5, n = x)
+#'   truncation_time = as.Date("2023-03-01")
 #' )
 truncate_linelist <- function(linelist,
-                              delay = function(x) stats::rlnorm(n = x, meanlog = 0.58, sdlog = 0.47), # nolint line_length_linter
-                              max_date = NULL,
-                              truncation_event = c("reporting", "onset",
-                                                   "admission", "outcome")) {
+                              truncation_time = 14,
+                              unit = c("days", "weeks", "months", "years"),
+                              direction = c("backwards", "forwards")) {
+  arg_ignore <- missing(unit) && missing(direction)
   .check_linelist(linelist)
```

Comment:
Haven't looked inside `check_linelist` but perhaps check that `date_reporting` doesn't have NAs (as otherwise the function will fail)?

Reply:
Resolved in ee15232c0d134f081858836ac471e08617e4529b.